modules/apps/transfer/keeper/relay.go

```diff
@@ -163,11 +163,13 @@ func (k Keeper) SendTransfer(
 	}
 
 	defer func() {
-		telemetry.SetGaugeWithLabels(
-			[]string{"tx", "msg", "ibc", "transfer"},
-			float32(token.Amount.Int64()),
-			[]metrics.Label{telemetry.NewLabel(coretypes.LabelDenom, fullDenomPath)},
-		)
+		if token.Amount.IsInt64() {
+			telemetry.SetGaugeWithLabels(
+				[]string{"tx", "msg", "ibc", "transfer"},
+				float32(token.Amount.Int64()),
+				[]metrics.Label{telemetry.NewLabel(coretypes.LabelDenom, fullDenomPath)},
+			)
+		}
```

Comment:
what happens if we're above int64? is it possible to just emit the string or a big Int?

Reply:
The amount isn't emitted in telemetry. The telemetry API requires a float32 which means nothing > int64 is supported and we cannot emit string or big.Int. This logic was taken from the SDK. The only other alternative is to remove the telemetry entirely